tools/online_editor/src/editor_widget.ts

```diff
@@ -0,0 +1,551 @@
+            source: @image-url("https://slint-ui.com/logo/slint-logo-full-light.svg");
+            height: 100px;
+        }
+        HorizontalLayout { alignment: center; Button { text: "OK!"; } }
+    }
+}
+`;
+
+function createModel(source: string): monaco.editor.ITextModel {
+  return monaco.editor.createModel(source, "slint");
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+(self as any).MonacoEnvironment = {
+  getWorker(_: unknown, _label: unknown) {
+    return new Worker(new URL("worker/monaco_worker.mjs", import.meta.url), {
+      type: "module",
+    });
+  },
+};
+
+function tabTitleFromURL(url: string): string {
+  if (url === "") {
+    return "unnamed.slint";
+  }
+  try {
+    const parsed_url = new URL(url);
+    const path = parsed_url.pathname;
+    return path.substring(path.lastIndexOf("/") + 1);
+  } catch (e) {
+    return url;
+  }
+}
+
+class EditorPaneWidget extends Widget {
+  private _auto_compile = true;
+  private _style = "fluent";
+  private _editor_documents: Map<string, ModelAndViewState>;
+  private _editor: monaco.editor.IStandaloneCodeEditor | undefined;
+  private _keystroke_timeout_handle: number | undefined;
+  private _base_url: string | undefined;
+
+  private _onRenderRequest?: (
+    _style: string,
+    _source: string,
+    _url: string,
+    _fetch: (_url: string) => Promise<string>
+  ) => Promise<monaco.editor.IMarkerData[]>;
+
+  private _onModelRemoved?: (_url: string) => void;
+  private _onModelAdded?: (_url: string) => void;
+  private _onModelSelected?: (_url: string) => void;
+  private _onModelsCleared?: () => void;
+
+  static createNode(): HTMLElement {
+    const node = document.createElement("div");
+    const content = document.createElement("div");
+    node.appendChild(content);
+
+    return node;
+  }
+
+  constructor() {
+    const node = EditorPaneWidget.createNode();
+
+    super({ node: node });
+    this._editor_documents = new Map();
+    this.setFlag(Widget.Flag.DisallowLayout);
+    this.addClass("content");
+    this.addClass("editor");
+    this.title.label = "Editor";
+    this.title.closable = false;
+    this.title.caption = `Slint code editor`;
+
+    this.setup_editor(this.contentNode);
+  }
+
+  protected get contentNode(): HTMLDivElement {
+    return this.node.getElementsByTagName("div")[0] as HTMLDivElement;
+  }
+
+  get current_editor_content(): string {
+    return this._editor?.getModel()?.getValue() || "";
+  }
+
+  compile() {
+    this.update_preview();
+  }
+
+  set auto_compile(value: boolean) {
+    this._auto_compile = value;
+  }
+
+  get auto_compile() {
+    return this._auto_compile;
+  }
+
+  set style(value: string) {
+    this._style = value;
+    this.update_preview();
+  }
+
+  get style() {
+    return this._style;
+  }
+
+  public clear_models() {
+    console.log("Clearing all models!");
+    this._editor_documents.clear();
+    if (this._onModelsCleared != null) {
+      this._onModelsCleared();
+    }
```

Comment:
I think that you can write this using the idiomatic optional chaining: `this._onModelsCleared?.();`